### ISSUE TITLE
Google\Client vs. Google_Client

### ISSUE THREAD

**Issue by wrufeger** [googleapis/google-api-php-client#1986]
Recent versions need initialization as Google\Client, but many Services still need Google_Client class ? 
Is there a difference and if, how to use both ?

**Comment by jdpedrie** [comment]
Prefer `Google\Client`. `Google_Client` is still available via an alias, so all services will continue working as normal.

**Comment by wrufeger** [comment]
Using `Google\Client` will throw type warnings if using most services as there `Google_Client` is still used. So these can be ignored then ?

**Comment by jdpedrie** [comment]
Here is a simple example showing the essentials of how the aliases work:

```php
<?php

class A {}
class_alias("A", "B");

// for code completion.
if (false) {
    class B extends A {}
}

function impl(A $foo) {
    var_dump(is_a($foo, "A"));
}

impl(new B);
```

The `if (false) {}` block is added in #1984, which should be part of the next release, hopefully next week. If you're seeing errors it may be because your IDE does not know about the alias. Despite this, everything should work as expected when the code is executed. If you are having issues other than the one described in the linked issue, please provide an example!

**Comment by bshaffer** [comment]
@wrufeger where are you seeing warnings? This should not be the case. Can you provide us with an example and the warning message you're receiving?

**Comment by bastomiadi** [comment]
In yii2 inget error if using Google\Client , how to solve this? 
![FB_IMG_1605375417101](https://user-images.githubusercontent.com/13336571/99153525-a998f880-26db-11eb-9893-19a6a4f1bbd5.jpg)

**Comment by igorsuhinin** [comment]
@bastomiadi Seems like it's a bug in 2.8.* release. I've just stumbled upon this as well.

The easiest solution would be to install the version 2.7.* instead of the latest one:

```
composer remove google/apiclient

composer install google/apiclient 2.7.*
```

Looks like everything works correctly after installing the previous version.

**Comment by bshaffer** [comment]
The IDE issue should now be fixed in [`v2.8.3`](https://github.com/googleapis/google-api-php-client/releases/tag/v2.8.3). Please update to the latest version, and open a new issue if you continue to see this error!
